packages/neo-one-smart-contract-lib/src/crowdsale/CrowdsaleContract.ts

```diff
@@ -0,0 +1,78 @@
+import { Address, constant, createEventNotifier, Fixed, SmartContract } from '@neo-one/smart-contract';
+import { Ownable } from '../ownership/Ownable';
+
+const notifyTokensPurchased = createEventNotifier<Address, Address, Fixed<8>, Fixed<8>>(
+  'tokens_purchased',
+  'purchaser',
+  'beneficiary',
+  'NEO',
+  'qty tokens',
+);
+
+export function CrowdsaleContract() {
+  abstract class CrowdsaleContractClass extends Ownable(SmartContract) {
+    public readonly token: Address = this.initialCrowdsaleToken();
+    public readonly rate: Fixed<8> = this.initialCrowdsaleRate();
+    public readonly wallet: Address = this.initialCrowdsaleWallet();
+    protected mutableNeoRaised: Fixed<8> = 0;
+    protected mutableTotalSupply: Fixed<8> = 0;
+
+    @constant
+    public get neoRaised(): Fixed<8> {
+      return this.mutableNeoRaised;
+    }
+
+    @constant
+    public get totalSupply(): Fixed<8> {
+      return this.mutableTotalSupply;
+    }
+
+    public mintTokens(purchaser: Address, beneficiary: Address, amount: Fixed<8>): boolean {
```

Comment:
`mintTokens` shouldn't take any arguments. See the existing `ICO` implementation. (Yes, I'm aware this does not match how it's done in OpenZeppelin)

Reply:
Borrowed much of the logic from ICO
pushed checks-and-balances into a `protected` `issue()` These "checks" are used by mixins to check if the sale is open, if the buyer is whitelisted or not, capped or not, et'al